src/components/intent.ts

```diff
@@ -494,7 +494,7 @@ export class Intent {
      * addition to those that are automatically chosen.
      */
     public async join(roomId: string, viaServers?: string[]) {
-        await this._ensureJoined(roomId, false, viaServers);
+        await this._ensureJoined(roomId, true, viaServers);
```

Comment:
I think this might be the wrong solution. The cache is useful as bridges like IRC will often do joins multiple times for users if it's missing state, as fundamentally bridges are usually not expected to cache state.

I'd be happy if we used `this.opts.backingStore.setMembership` `this.opts.backingStore.getMembership` in `leave` though.